rand_core/src/os.rs

```diff
@@ -36,51 +36,53 @@ use getrandom::getrandom;
 ///
 /// # Usage example
 /// ```
-/// use rand_core::{RngCore, OsRng};
+/// use rand_core::{TryRngCore, OsRng};
 ///
 /// let mut key = [0u8; 16];
-/// OsRng.fill_bytes(&mut key);
-/// let random_u64 = OsRng.next_u64();
+/// OsRng.try_fill_bytes(&mut key).unwrap();
+/// let random_u64 = OsRng.try_next_u64().unwrap();
 /// ```
 ///
 /// [getrandom]: https://crates.io/crates/getrandom
 #[cfg_attr(doc_cfg, doc(cfg(feature = "getrandom")))]
 #[derive(Clone, Copy, Debug, Default)]
 pub struct OsRng;
 
-impl CryptoRng for OsRng {}
+impl TryRngCore for OsRng {
+    type Error = getrandom::Error;
```

Comment:
This exposes `getrandom::Error` as part of public API. I think it's fine since `getrandom` is quite stable and we no longer need boxing provided by `rand_core::Error`.

Reply:
A caveat is that we can't release `rand_core` v1.0 before `getrandom` v1.0, but maybe we shouldn't care about that.

I think otherwise this is perfectly fine.

Reply:
In some cases the semver trick could allow such migration without breaking changes.